### PR TITLE
[new doc] security model

### PR DESCRIPTION
**Pull request by pdg744**
This PR adds a new doc to help users understand the end-to-end proving stack and the associated security model. 
The doc is included in both `next` and `0.21`

### CODE REVIEW

File: website/api/security-model.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+## Zero-Knowledge Proving
+
+Proofs generated by the Recursion Prover and proofs generated by the STARK-to-SNARK Prover reveal no information beyond the contents of the [receipt claim].
+Even with unlimited computing power, no secrets can be extracted from proofs posted on-chain.
+
+**Warning 1**: Whoever is generating the proofs can see the secret data. Users may choose to use any of these provers locally to ensure their data stays private.
+
+**Warning 2:** Proofs generated by the **RISC-V Prover** that haven't been passed through the **Recursion Prover** leak information about the length of execution.
+Passing proofs through the Recursion Prover resolves this warning: recursion proofs leak no information about execution length.
```
Comment (tzerrell) [inline website/api/security-model.md:45]: Perhaps format these warnings as [admonitions](https://docusaurus.io/docs/next/markdown-features/admonitions)?

```suggestion
:::warning

Whoever is generating the proofs can see the secret data. Users may choose to use any of these provers locally to ensure their data stays private.

:::

:::warning

Proofs generated by the **RISC-V Prover** that haven't been passed through the **Recursion Prover** leak information about the length of execution.
Passing proofs through the Recursion Prover resolves this warning: recursion proofs leak no information about execution length.

:::
```

File: website/api/security-model.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+We estimate that it would cost over 400 billion USD of computational resources to construct a fake proof for one of our STARK circuits.
+
+These estimates are based on the approach used in [this article by Justin Thaler].
+The napkin math here is as follows:
+
+- Thaler estimated $1.6 million to launch a viable attack against an 80 bit STARK system.
+- 98 bits is $2^{18}$ times more secure than 80 bits.
+
+The analysis for bits of security for the RISC-V and Recursion Prover can be found in the [security calculator].
```
Comment (tzerrell) [inline website/api/security-model.md:80]: The `[security calculator]` link is broken

**Review by tzerrell — COMMENTED** [review]
:tada:

File: website/api/security-model.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+[control ID]: ../../terminology#control-id
+[control root]: ../../terminology#control-root
+[execution trace]: ../../terminology#execution-trace
+[ethSTARK documentation]: https://eprint.iacr.org/2021/582
+[deterministically]: ../../terminology#deterministic-builds
+[GitHub issue from Zcash]: https://github.com/zcash/zcash/issues/714
+[receipt claim]: ../../terminology#receipt-claim
+[RISC Zero zkVM: Scalable, Transparent Arguments of RISC-V Integrity]: ./proof-system-in-detail.pdf
+[RISC Zero zkVM]: ../zkvm
+[this article by Justin Thaler]: https://a16zcrypto.com/posts/article/snark-security-and-performance/
```
Comment (pdg744) [inline website/api/security-model.md:114]:
```suggestion
[security calculator]: https://github.com/risc0/risc0/blob/main/risc0/zkp/src/prove/soundness.rs
[this article by Justin Thaler]: https://a16zcrypto.com/posts/article/snark-security-and-performance/
```

File: website/api_versioned_docs/version-0.21/security-model.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+[control ID]: ../terminology#control-id
+[control root]: ../terminology#control-root
+[execution trace]: ../terminology#execution-trace
+[ethSTARK documentation]: https://eprint.iacr.org/2021/582
+[deterministically]: ../terminology#deterministic-builds
+[GitHub issue from Zcash]: https://github.com/zcash/zcash/issues/714
+[receipt claim]: ../terminology#receipt-claim
+[RISC Zero zkVM: Scalable, Transparent Arguments of RISC-V Integrity]: ./proof-system-in-detail.pdf
+[RISC Zero zkVM]: ./zkvm
+[this article by Justin Thaler]: https://a16zcrypto.com/posts/article/snark-security-and-performance/
```
Comment (pdg744) [inline website/api_versioned_docs/version-0.21/security-model.md:106]:
```suggestion
[security calculator]: https://github.com/risc0/risc0/blob/main/risc0/zkp/src/prove/soundness.rs
[this article by Justin Thaler]: https://a16zcrypto.com/posts/article/snark-security-and-performance/
```

File: website/api/security-model.md (hunk 2 above)
Comment (pdg744) [reply website/api/security-model.md:88]: good catch -- should be fixed now

**Comment by pdg744** [comment]
Rad to see that CI includes a check to make sure we're pointing to stable releases... in this case, the calculator exists on main but not on `0.21`, so I think it probably makes sense to override. Alternatively, could update the link to point to the PR instead.

**Review by tzerrell — APPROVED** [review]

**Comment by pdg744** [comment]
Planning to change the "how secure is the STARK section" from a cost estimate to a time-based estimate on a follow-up PR, but this is good enough for publication as is.

**Comment by flaub** [comment]
How did this land when CI checks failed?

**Comment by pdg744** [comment]
weird. 

worth noting that i think the thing that landed should not have failed CI checks -- the test references a pointer to main, but i believe the last commit on the PR removed that pointer...

**Pull request merged**